storey/table.py

```diff
@@ -244,6 +288,151 @@ def __getitem__(self, key):
+                    received_job_count += 1
+                    if task is _termination_obj:
+                        if received_job_count in self_sent_jobs:
+                            await self._q.put(_termination_obj)
+                            continue
+                        for key, pending_event in self._pending_by_key.items():
+                            if pending_event.pending and not pending_event.in_flight:
+                                for job in pending_event.pending:
+                                    resp = await self._internal_persist_key(key, job.data)
+                                    self._changed_keys.discard(key)
+                                    if job.callback:
+                                        await job.callback(job.extra_data, resp)
+                        break
+
+                job = task[0]
+                completed = await task[1]
+
+                if isinstance(job.key, list):
+                    job_key = str(job.key)
+                else:
+                    job_key = job.key
+
+                for done_job in self._pending_by_key[job_key].in_flight:
+                    if len(self._pending_by_key[job_key].pending) == 0:
+                        self._changed_keys.discard(job_key)
+                    if done_job.callback:
+                        await done_job.callback(done_job.extra_data, completed)
+                self._pending_by_key[job_key].in_flight = []
+
+                # If we got more pending events for the same key process them
+                if self._pending_by_key[job_key].pending:
+                    self._pending_by_key[job_key].in_flight = self._pending_by_key[job_key].pending
+                    self._pending_by_key[job_key].pending = []
+
+                    future_task = self._safe_process_events(self._pending_by_key[job_key].in_flight)
+                    tail_position = received_job_count + self._q.qsize()
+                    jobs_at_tail = self_sent_jobs.get(tail_position, [])
+                    jobs_at_tail.append((job, asyncio.get_running_loop().create_task(future_task)))
+                    self_sent_jobs[tail_position] = jobs_at_tail
+                else:
+                    del self._pending_by_key[job_key]
+        except BaseException as ex:
+            if task and task is not _termination_obj:
+                self._changed_keys.discard(task[0].key)
+                if task[0].extra_data and task[0].extra_data._awaitable_result:
+                    task[0].extra_data._awaitable_result._set_error(ex)
+            if not self._q.empty():
+                await self._q.get()
+            raise ex
+
+    async def _terminate(self):
+        if self._flush_exception is not None:
+            raise self._flush_exception
+        if not self._terminated:
+            self._terminated = True
+            for key in self._changed_keys:
+                await self._persist(_PersistJob(key, None, None), from_terminate=True)
```

Comment:
Do we really want to persist them serially?

Reply:
It may also persist twice, here and in the flush worker, because the flush worker is only cancelled after this bit.

Reply:
yea changed it, i first cancel flush worker and then continue with terminate flow